File: OOP/06.Encapsulation and Polymorphism/01.Shapes/Circle.cs
namespace Shape
{
    using System;

    public class Circle : IShape
    {
        private double radius;

        public Circle(double radius)
        {
            this.Radius = radius;
        }

        public double Radius
        {
            get
            {
                return this.radius;
            }

            set
            {
                if (value <= 0)
                {
                    throw new ArgumentOutOfRangeException("value", "Radius cannot be zero or negative.");
                }

                this.radius = value;
            }
        }

        public double CalculateArea()
        {
            return Math.PI * (2 * this.Radius);
        }

        public double CalculatePerimeter()
        {
            return 2 * Math.PI * this.Radius;
        }
    }
}

File: OOP/08.OOP in Java/src/com/softuni/oop/geometry/Circle.java
package com.softuni.oop.geometry;

public class Circle extends PlaneShape {
    private double radius;

    public Circle(Point2D center, double radius) {
        super(center);
        this.setRadius(radius);
    }

    public double getRadius() {
        return this.radius;
    }

    private void setRadius(double radius) {
        if (radius <= 0.0) {
            throw new IllegalArgumentException("Invalid radius - cannot be of negative or zero value.");
        }

        this.radius = radius;
    }

    @Override
    public double getArea() {
        return Math.PI * this.getRadius() * this.getRadius();
    }

    @Override
    public double getPerimeter() {
        return 2 * Math.PI * this.getRadius();
    }

    @Override
    public String toString() {
        StringBuilder output = new StringBuilder();
        output.append("Shape type: Circle\n");
        output.append(super.toString());

        return output.toString();
    }
}
